integration/benchmark/worker_stats_tracker_test.go

```diff
@@ -1,25 +1,30 @@
 package benchmark
 
 import (
+	"context"
 	"testing"
-	"time"
 
 	"github.com/stretchr/testify/assert"
 )
 
 // TestWorkerStatsTracker creates a new worker stats,
 // adds a couple of transactions there and verifies that average tps is correct.
 func TestWorkerStatsTracker(t *testing.T) {
-	st := NewWorkerStatsTracker()
-	st.StartPrinting(time.Second)
-	defer st.StopPrinting()
-
+	st := NewWorkerStatsTracker(context.Background())
 	st.AddWorkers(1)
 
-	startTime := time.Now()
-	endTime := startTime.Add(time.Second)
-	assert.Equal(t, 0., st.AvgTPSBetween(startTime, endTime))
-	st.AddTxSent()
-	st.AddTxSent()
-	assert.Equal(t, 2., st.AvgTPSBetween(startTime, endTime))
+	stats := st.getStats()
+	assert.Equal(t, 0, stats.txsSent)
```

Comment:
Shouldn't this test use `require.Equal()` instead of `assert.Equal()`? Otherwise the test won't fail if the values don't match.

Reply:
It should fail at the end, but will continue to the end instead of failing fast.